File: src/aquadig/Mid.java
package aquadig;

import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;
import java.io.InputStream;
import java.util.Random;
import javax.microedition.lcdui.Display;
import javax.microedition.lcdui.Image;
import javax.microedition.media.Manager;
import javax.microedition.media.Player;
import javax.microedition.midlet.MIDlet;
import javax.swing.Timer;

public final class Mid extends MIDlet {
   static byte a = 100;
   static d b = null;
   private static Display q = null;
   static c c;
   public static int frameMs = 25;
   private static final long s = 1000 / frameMs;
   private static long t;
   /** Logical game time (ms); incremented each frame by {@link #s}. */
   public static volatile long clockMs;
   private Timer gameTimer;
   static boolean f = true;
   static Random rng = new Random();
   public static com.vlm.redstar.client.core.csp.a h;
   static Player i = null;
   static final String[] j = new String[]{"/res/theme.mid", "/res/credits.mid", "/res/shopping.mid", "/res/death.mid", "/res/win.mid"};
   static byte k = -1;
   final String l;
   final String m;
   public static boolean n = false;
   public long o = 0L;
   public boolean p = false;

   public Mid() {
      this.l = this.getAppProperty("MIDlet-Version");
      this.m = f();
      a = 100;
   }

   public final void startApp() {
      if (gameTimer == null) {
         q = Display.getDisplay(this);
         b = new d(this);
         q.setCurrent(b);
         f = true;
         clockMs = 0L;
         gameTimer = new Timer((int) s, new GameTick(this));
         gameTimer.setInitialDelay(0);
         gameTimer.setRepeats(true);
         gameTimer.start();
      }
   }

   /**
    * EDT frame tick (Swing {@link Timer}). Avoids {@link Thread#sleep}, which SwingJS does not
    * support; browser timing uses the same path as other Swing timers.
    */
   private void tickFrame() {
      if (!f) {
         return;
      }
      try {
         t = clockMs;
         this.c();
         clockMs += s;
         b.repaint();
         b.serviceRepaints();
      } catch (Throwable ignored) {
      }
   }

   private static final class GameTick implements ActionListener {
      private final Mid mid;

      GameTick(Mid mid) {
         this.mid = mid;
      }

      @Override
      public void actionPerformed(ActionEvent e) {
         mid.tickFrame();
      }
   }

   public final void pauseApp() {
   }

   private void c() {
      try {
         switch (a) {
            case 100:
               if (com.vlm.redstar.client.ui.d.b()) {
                  b.L.c();
                  ParticleTick.tick();
               }
               break;
            case 101:
               if (!com.vlm.redstar.client.ui.a.a()) {
                  tickIdle();
               }
               break;
            case 116:
               b.o();
               break;
         }

         if (clockMs >= this.o + 400L) {
            n = !n;
            this.o = clockMs;
         }
      } catch (Throwable var2) {
      }
   }

   private static void tickIdle() {
      if (d.e != null) {
         if (d.f > 0L) {
            d.e.g_ = d.e.g_ + (t - d.f);
            d.f = 0L;
         } else {
            d.e.g_ = d.e.g_ + s;
         }
      }

      e.p();
      g.a();
      ParticleTick.tick();
   }

   private static void e() {
      if (h != null) {
         com.vlm.redstar.client.core.csp.a.a();
      }

      ParticleTick.reset();
      if (c != null) {
         c.a();
         c = null;
      }

      if (b != null) {
         b.f();
         b = null;
      }

      System.gc();
   }

   public final void destroyApp(boolean var1) {
      f = false;
      if (gameTimer != null) {
         gameTimer.stop();
         gameTimer = null;
      }
      e();
      this.notifyDestroyed();
   }

   static Image a(String var0) {
      Image var1 = null;

      try {
         var1 = Image.createImage(var0);
      } catch (Throwable var3) {
      }

      return var1;
   }

   private static void a(byte var0) {
      try {
         byte var2 = a;
         a = 108;
         b.af = 1;
         b.ae = 1;
         b.repaint();
         b.serviceRepaints();
         i = a(j[var0], "audio/midi");
         a = var2;
      } catch (Throwable var3) {
      }
   }

   private static Player a(String var0, String var1) {
      /*Player var2 = null;
      Object var3 = new Object();

      try {
         var2 = Manager.createPlayer(var3.getClass().getResourceAsStream(var0), var1);
      } catch (Throwable var5) {
      }

      return var2;*/

      return null;
   }

   public static void a(byte var0, int var1) {
      if (com.vlm.redstar.client.ui.d.d) {
         try {
            label27: {
               if (i != null && k != var0) {
                  a();
               } else if (i != null) {
                  break label27;
               }

               a(var0);
            }

            i.realize();
            i.prefetch();
            i.setLoopCount(var1);
            i.start();
            k = var0;
         } catch (Throwable var3) {
         }
      }
   }

   static void a() {
      try {
         if (i != null) {
            if (i.getState() == 400) {
               i.stop();
            }

            i.deallocate();
            i.close();
            i = null;
            k = -1;
            System.gc();
         }
      } catch (Throwable var1) {
      }
   }

   static int a(int var0) {
      return Math.abs(rng.nextInt() % var0);
   }

   static int b(int var0) {
      return rng.nextInt() % var0;
   }

   static boolean b() {
      return rng.nextInt() >= 0;
   }

   static boolean a(int var0, int var1, int var2, int var3, int var4, int var5, int var6, int var7) {
      if (var1 + var3 < var5) {
         return false;
      } else if (var1 > var5 + var7) {
         return false;
      } else {
         return var0 + var2 < var4 ? false : var0 <= var4 + var6;
      }
   }

   public static int a(int var0, int var1) {
      int var5 = var0 >>> 24;
      int var6 = 256 - var5;
      return 0xFF0000 & (var0 & 0xFF0000) * var5 + (var1 & 0xFF0000) * var6 >> 8
         | 0xFF00 & (var0 & 0xFF00) * var5 + (var1 & 0xFF00) * var6 >> 8
         | 0xFF & (var0 & 0xFF) * var5 + (var1 & 0xFF) * var6 >> 8;
   }

   private static String f() {
      byte[] var0 = new byte[0];

      try {
         InputStream var1;
         var0 = new byte[(var1 = new Object().getClass().getResourceAsStream("/res/version.txt")).available()];
         var1.read(var0, 0, var1.available());
         var1.close();
      } catch (Throwable var2) {
      }

      return new String(var0);
   }

   static void c(int var0) {
      if (com.vlm.redstar.client.ui.d.e) {
         q.vibrate(var0);
      }
   }
}

File: docs/swingjs/j2s/aquadig/Mid.js
(function(){var P$=Clazz.newPackage("aquadig"),p$1={},I$=[[0,'java.util.Random','javax.microedition.lcdui.Display','aquadig.d','javax.swing.Timer',['aquadig.Mid','.GameTick'],'com.vlm.redstar.client.ui.d','aquadig.ParticleTick','com.vlm.redstar.client.ui.a','aquadig.e','aquadig.g','com.vlm.redstar.client.core.csp.a','aquadig.c','javax.microedition.lcdui.Image']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "Mid", function(){
Clazz.newInstance(this, arguments,0,C$);
}, 'javax.microedition.midlet.MIDlet');
C$.$classes$=[['GameTick',26]];

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
this.o=0;
this.p=false;
},1);

C$.$fields$=[['Z',['p'],'J',['o'],'S',['l','m'],'O',['gameTimer','javax.swing.Timer']]
,['Z',['f','n'],'B',['a','k'],'I',['frameMs'],'J',['s','t','clockMs'],'O',['b','aquadig.d','q','javax.microedition.lcdui.Display','c','aquadig.c','rng','java.util.Random','h','com.vlm.redstar.client.core.csp.a','i','javax.microedition.media.Player','j','String[]']]]

Clazz.newMeth(C$, 'c$',  function () {
Clazz.super_(C$, this);
this.l=this.getAppProperty$S("MIDlet-Version");
this.m=C$.f$();
C$.a=($b$[0] = 100, $b$[0]);
}, 1);

Clazz.newMeth(C$, 'startApp$',  function () {
if (this.gameTimer == null ) {
C$.q=$I$(2).getDisplay$javax_microedition_midlet_MIDlet(this);
C$.b=Clazz.new_($I$(3,1).c$$aquadig_Mid,[this]);
C$.q.setCurrent$javax_microedition_lcdui_Displayable(C$.b);
C$.f=true;
C$.clockMs=0;
this.gameTimer=Clazz.new_([Long.$ival(C$.s), Clazz.new_($I$(5,1).c$$aquadig_Mid,[this])],$I$(4,1).c$$I$java_awt_event_ActionListener);
this.gameTimer.setInitialDelay$I(0);
this.gameTimer.setRepeats$Z(true);
this.gameTimer.start$();
}});

Clazz.newMeth(C$, 'tickFrame',  function () {
if (!C$.f) {
return;
}try {
C$.t=C$.clockMs;
p$1.c.apply(this, []);
(C$.clockMs=Long.$add(C$.clockMs,(C$.s)));
C$.b.repaint$();
C$.b.serviceRepaints$();
} catch (ignored) {
}
}, p$1);

Clazz.newMeth(C$, 'pauseApp$',  function () {
});

Clazz.newMeth(C$, 'c',  function () {
try {
switch (C$.a) {
case 100:
if ($I$(6).b$()) {
C$.b.L.c$$();
$I$(7).tick$();
}break;
case 101:
if (!$I$(8).a$()) {
C$.tickIdle$();
}break;
case 116:
C$.b.o$();
break;
}
if (Long.$ge(C$.clockMs,Long.$add(this.o,400) )) {
C$.n=!C$.n;
this.o=C$.clockMs;
}} catch (var2) {
}
}, p$1);

Clazz.newMeth(C$, 'tickIdle$',  function () {
if ($I$(3).e != null ) {
if (Long.$gt($I$(3).f,0 )) {
$I$(3).e.g_=Long.$add($I$(3).e.g_,(Long.$sub(C$.t,$I$(3).f)));
$I$(3).f=0;
} else {
$I$(3).e.g_=Long.$add($I$(3).e.g_,C$.s);
}}$I$(9).p$();
$I$(10).a$();
$I$(7).tick$();
}, 1);

Clazz.newMeth(C$, 'e$',  function () {
if (C$.h != null ) {
$I$(11).a$();
}$I$(7).reset$();
if (C$.c != null ) {
$I$(12).a$();
C$.c=null;
}if (C$.b != null ) {
C$.b.f$();
C$.b=null;
}System.gc$();
}, 1);

Clazz.newMeth(C$, 'destroyApp$Z',  function (var1) {
C$.f=false;
if (this.gameTimer != null ) {
this.gameTimer.stop$();
this.gameTimer=null;
}C$.e$();
this.notifyDestroyed$();
});

Clazz.newMeth(C$, 'a$S',  function (var0) {
var var1=null;
try {
var1=$I$(13).createImage$S(var0);
} catch (var3) {
}
return var1;
}, 1);

Clazz.newMeth(C$, 'a$B',  function (var0) {
try {
var var2=C$.a;
C$.a=($b$[0] = 108, $b$[0]);
C$.b.af=1;
C$.b.ae=1;
C$.b.repaint$();
C$.b.serviceRepaints$();
C$.i=C$.a$S$S(C$.j[var0], "audio/midi");
C$.a=var2;
} catch (var3) {
}
}, 1);

Clazz.newMeth(C$, 'a$S$S',  function (var0, var1) {
return null;
}, 1);

Clazz.newMeth(C$, 'a$B$I',  function (var0, var1) {
if ($I$(6).d) {
try {
 label27 : {
if (C$.i != null  && C$.k != var0 ) {
C$.a$();
} else if (C$.i != null ) {
break label27;
}C$.a$B(var0);
}C$.i.realize$();
C$.i.prefetch$();
C$.i.setLoopCount$I(var1);
C$.i.start$();
C$.k=var0;
} catch (var3) {
}
}}, 1);

Clazz.newMeth(C$, 'a$',  function () {
try {
if (C$.i != null ) {
if (C$.i.getState$() == 400) {
C$.i.stop$();
}C$.i.deallocate$();
C$.i.close$();
C$.i=null;
C$.k=($b$[0] = -1, $b$[0]);
System.gc$();
}} catch (var1) {
}
}, 1);

Clazz.newMeth(C$, 'a$I',  function (var0) {
return Math.abs(C$.rng.nextInt$() % var0);
}, 1);

Clazz.newMeth(C$, 'b$I',  function (var0) {
return C$.rng.nextInt$() % var0;
}, 1);

Clazz.newMeth(C$, 'b$',  function () {
return C$.rng.nextInt$() >= 0;
}, 1);

Clazz.newMeth(C$, 'a$I$I$I$I$I$I$I$I',  function (var0, var1, var2, var3, var4, var5, var6, var7) {
if (var1 + var3 < var5) {
return false;
} else if (var1 > var5 + var7) {
return false;
} else {
return var0 + var2 < var4 ? false : var0 <= var4 + var6;
}}, 1);

Clazz.newMeth(C$, 'a$I$I',  function (var0, var1) {
var var5=var0 >>> 24;
var var6=256 - var5;
return 16711680 & (var0 & 16711680) * var5 + (var1 & 16711680) * var6 >> 8 | 65280 & (var0 & 65280) * var5 + (var1 & 65280) * var6 >> 8 | 255 & (var0 & 255) * var5 + (var1 & 255) * var6 >> 8;
}, 1);

Clazz.newMeth(C$, 'f$',  function () {
var var0=Clazz.array(Byte.TYPE, [0]);
try {
var var1;
var0=Clazz.array(Byte.TYPE, [(var1= Clazz.new_().getClass$().getResourceAsStream$S("/res/version.txt")).available$()]);
var1.read$BA$I$I(var0, 0, var1.available$());
var1.close$();
} catch (var2) {
}
return  String.instantialize(var0);
}, 1);

Clazz.newMeth(C$, 'c$I',  function (var0) {
if ($I$(6).e) {
C$.q.vibrate$I(var0);
}}, 1);

C$.$static$=function(){C$.$static$=0;
C$.a=($b$[0] = 100, $b$[0]);
C$.b=null;
C$.q=null;
C$.frameMs=25;
C$.s=(1000/C$.frameMs|0);
C$.f=true;
C$.rng=Clazz.new_($I$(1,1));
C$.i=null;
C$.j=Clazz.array(String, -1, ["/res/theme.mid", "/res/credits.mid", "/res/shopping.mid", "/res/death.mid", "/res/win.mid"]);
C$.k=($b$[0] = -1, $b$[0]);
C$.n=false;
};
var $b$ = new Int8Array(1);
;
(function(){/*c*/var C$=Clazz.newClass(P$.Mid, "GameTick", function(){
Clazz.newInstance(this, arguments[0],false,C$);
}, null, 'java.awt.event.ActionListener');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
},1);

C$.$fields$=[['O',['mid','aquadig.Mid']]]

Clazz.newMeth(C$, 'c$$aquadig_Mid',  function (mid) {
;C$.$init$.apply(this);
this.mid=mid;
}, 1);

Clazz.newMeth(C$, 'actionPerformed$java_awt_event_ActionEvent',  function (e) {
p$1.tickFrame.apply(this.mid, []);
});

Clazz.newMeth(C$);
})()
})();
;Clazz.setTVer('3.3.1-v5');//Created 2026-06-25 15:29:03 Java2ScriptVisitor version 3.3.1-v5 net.sf.j2s.core.jar version 3.3.1-v5
